File: usvm-jvm-spring/usvm-jvm-spring-api/src/main/java/org/usvm/spring/query/specification/ByExpression.java
package org.usvm.spring.query.specification;

import org.usvm.spring.query.expression.AExpression;

public class ByExpression extends ASpecification {
    public AExpression expr;

    public Object translateMethod = null;

    public ByExpression() {
    }

    public ByExpression(AExpression expr) {
        this.expr = expr;
    }

    @Override
    public <R, C> R accept(ISpecificationVisitor<R, C> visitor, C ctx) {
        return visitor.visit(this, ctx);
    }
}

File: usvm-jvm-spring/src/main/kotlin/machine/interpreter/transformers/springjpa/query/specification/ByExpression.kt
package machine.interpreter.transformers.springjpa.query.specification

import machine.interpreter.transformers.springjpa.query.CommonInfo
import machine.interpreter.transformers.springjpa.query.expression.toLambda
import org.jacodb.api.jvm.JcMethod
import org.usvm.spring.query.specification.ByExpression

fun ByExpression.getTranslateMethod(info: CommonInfo): JcMethod {
    translateMethod?.also { return it as JcMethod }
    translateMethod = expr.toLambda(info)
    return translateMethod!! as JcMethod
}
